src/libs/actions/Session.js

```diff
@@ -139,8 +139,9 @@ function signIn(password, exitTo, twoFactorAuthCode) {
                     setSuccessfulSignInData(createLoginResponse, exitTo);
 
                     // If we have an old login for some reason, we should delete it before storing the new details
-                    if (credentials.login) {
```

Comment:
It didn't make sense to me to delete an old login if we have the `login` without `password`. Also, we set the login when calling `GetAccountStatus` and that doesn't seem like the login we want to delete.

Reply:
I don't think it's necessary to add a password check, since the password isn't used as part of the request it doesn't matter if there is a password saved to Onyx or not.

Reply:
>the password isn't used as part of the request

Didn't understand that part. Where does this happen?

We are adding a login to credentials before this is called so won't we always have a `login`? e.g. if you set your email in step one then it is there and we'll try to delete it after you authenticate?

Reply:
Actually no worries. I think this should be resolved by your other comment.